From c3310e432345d9d3ca5323cfc626fb4d46009d3b Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Fri, 14 Apr 2023 20:05:25 +0000
Subject: GCC: fix ambiguous Observer type in //net

GCC sees two definitions of Observer for AddObserver and
RemoveObserver arguments. Explicitly select the right one.

Bug: 819294
Change-Id: I55d7b4afb98f0bebd35051d54588f2f9d0390966
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4408875
Reviewed-by: Matt Mueller <mattm@chromium.org>
Commit-Queue: Stephan Hartmann <stha09@googlemail.com>
Cr-Commit-Position: refs/heads/main@{#1130642}
---
 net/cert/coalescing_cert_verifier.cc      | 4 ++--
 net/cert/coalescing_cert_verifier.h       | 4 ++--
 net/cert/trial_comparison_cert_verifier.h | 4 ++--
 3 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/net/cert/coalescing_cert_verifier.cc b/net/cert/coalescing_cert_verifier.cc
index ef3622baba585..350f964323418 100644
--- a/net/cert/coalescing_cert_verifier.cc
+++ b/net/cert/coalescing_cert_verifier.cc
@@ -433,11 +433,11 @@ void CoalescingCertVerifier::SetConfig(const CertVerifier::Config& config) {
   IncrementGenerationAndMakeCurrentJobsUnjoinable();
 }
 
-void CoalescingCertVerifier::AddObserver(Observer* observer) {
+void CoalescingCertVerifier::AddObserver(CertVerifier::Observer* observer) {
   verifier_->AddObserver(observer);
 }
 
-void CoalescingCertVerifier::RemoveObserver(Observer* observer) {
+void CoalescingCertVerifier::RemoveObserver(CertVerifier::Observer* observer) {
   verifier_->RemoveObserver(observer);
 }
 
diff --git a/net/cert/coalescing_cert_verifier.h b/net/cert/coalescing_cert_verifier.h
index aa8a7fbeef9a4..3a0d0030ba773 100644
--- a/net/cert/coalescing_cert_verifier.h
+++ b/net/cert/coalescing_cert_verifier.h
@@ -46,8 +46,8 @@ class NET_EXPORT CoalescingCertVerifier : public CertVerifier,
              std::unique_ptr<CertVerifier::Request>* out_req,
              const NetLogWithSource& net_log) override;
   void SetConfig(const CertVerifier::Config& config) override;
-  void AddObserver(Observer* observer) override;
-  void RemoveObserver(Observer* observer) override;
+  void AddObserver(CertVerifier::Observer* observer) override;
+  void RemoveObserver(CertVerifier::Observer* observer) override;
 
   uint64_t requests_for_testing() const { return requests_; }
   uint64_t inflight_joins_for_testing() const { return inflight_joins_; }
diff --git a/net/cert/trial_comparison_cert_verifier.h b/net/cert/trial_comparison_cert_verifier.h
index 1712a81b0b1ee..d8a9329f9342b 100644
--- a/net/cert/trial_comparison_cert_verifier.h
+++ b/net/cert/trial_comparison_cert_verifier.h
@@ -89,8 +89,8 @@ class NET_EXPORT TrialComparisonCertVerifier
              std::unique_ptr<Request>* out_req,
              const NetLogWithSource& net_log) override;
   void SetConfig(const Config& config) override;
-  void AddObserver(Observer* observer) override;
-  void RemoveObserver(Observer* observer) override;
+  void AddObserver(CertVerifier::Observer* observer) override;
+  void RemoveObserver(CertVerifier::Observer* observer) override;
   void UpdateVerifyProcData(
       scoped_refptr<CertNetFetcher> cert_net_fetcher,
       const CertVerifyProcFactory::ImplParams& impl_params) override;
-- 
2.37.2

